icefoganalytics/travel-authorization
# Standardize Project Structure Based On Services

Fixes https://github.com/icefoganalytics/travel-authorization/issues/23

# Context

The current project structure is set up based on the concept of "source" and "non-source" code. This confuses various tooling defaults, as most tools expect a different layout. i.e. Typescript expects all code to have the package.json then a /src folder, instead of the package.json being inside the /src folder.

A more flexible, and standard approach is to separate the code base on "services" then "source" vs. "config" code.

e.g.
instead of
```bash
src/
|
|-- api/
|
|-- web/
```

Change the project structure to
```bash
project-root-directory/
|
|-- api/
|   |-- src/
|   |-- various config files for the api service
|
|-- web/
|   |-- src/
|   |-- various config files for the web service
|
|-- db/
|   |-- various config files for the database service
|
|-- top level config files and production docker files
```

# Testing Instructions
1. Boot the database service via `dev up`
2. Boot the api service via `dev up_api`
3. Boot the web service via `dev up_web`
4. Check that the app boots correctly and that you can log in.
5. Check that the production build works as well by following the instructions in the README.
    > Note that the production build doesn't have the correct configuration, so you can't use the app yet, but it does appear to build correctly.
